File: src/chanhyun/week6/주차 요금 계산.py
def count_minutes(end, start):
    s = start.split(':')
    s_t = int(s[0])*60+int(s[1])
    e = end.split(':')
    e_t = int(e[0])*60+int(e[1])
    return e_t-s_t

def count_fee(m, fees):
    money = fees[1]
    if m > fees[0]:
        if (m - fees[0])%fees[2] == 0:
            money += (m - fees[0])//fees[2] * fees[3]
        else:
            money += ((m - fees[0])//fees[2] + 1) * fees[3]
    return money

def solution(fees, records):
    answer = []
    book = dict()
    minutes = dict()
    for record in records:
        r = record.split()
        if r[1] in book:
            book[r[1]].append(r[0])
        else:
            book[r[1]] = [r[0]]
        if r[1] not in minutes:
            minutes[r[1]] = 0
    
    for k, v in book.items():
        if len(v) % 2 == 1:
            book[k].append("23:59")
    
    for k, v in book.items():
        for i in range(1, len(v), 2):
            minutes[k] += count_minutes(v[i], v[i-1])
    
    for k, v in minutes.items():
        answer.append((k, count_fee(v, fees)))
        
    answer.sort()
    result = []
    for a in answer:
        result.append(a[1])
    return result

File: src/taeeun/week6/주차 요금 계산.js
function solution(fees, records) {
    var answer = [];
    
    const [defaultTime, defaultFee, unitTime, unitFee]=fees;
    let parkinglot=[];
    

    const convertTime=(time)=>{
        const [hour, min]=time.split(":");
        return [parseInt(hour), parseInt(min)];
    }
    
    const calculateTime=(enter, out)=>{
        const hour=out[0]-enter[0];
        const min=out[1]-enter[1];
        return hour*60+min;
    }
    
    const calculateFees=(totalTime)=>{
        const fee=Math.ceil((totalTime-defaultTime)/unitTime)*unitFee;
        if (fee>0) return defaultFee+fee;
        else return defaultFee;
    }
    
    for (const record of records) {
        const [time, carNum, isEnter]=record.split(" ");

        const car=parkinglot.find((e)=>e.carNum===carNum);
        if (isEnter==="IN") {
            if (car===undefined) parkinglot.push({carNum: carNum, enter: time, totalTime:0 });
            else car.enter=time;
        }
        else {
            const enter=convertTime(car.enter);
            const out=convertTime(time);
            const totalTime=calculateTime(enter, out);
            car.enter='';
            car.totalTime+=totalTime;
        }
    }
    
    for (const cars of parkinglot) {
        //만약 출차내역이 없다면 23:59분에 출차
        if (cars.enter!=='') {
            const enter=convertTime(cars.enter);
            const out=[23, 59];
            const totalTime=calculateTime(enter, out);
            cars.totalTime+=totalTime;
        }
    }
    
    parkinglot.sort((a, b)=> a.carNum-b.carNum);
    
    for (const cars of parkinglot) {
        answer.push(calculateFees(cars.totalTime));
    }
    
    
    return answer;
}
